### PR TITLE
HUB-1396 : Added sanitization method to fix JS client privacy violation

### PR DESCRIPTION
**Pull request by Haripriya99j**
**Jira:**
https://cirium.atlassian.net/browse/HUB-1396

**Integration tests:**
https://ddt-jenkins.prod.flightstats.io/job/run-hub-integration-tests-against-cluster/116/console

**System tests:**
https://ddt-jenkins.prod.flightstats.io/job/run-hub-system-tests/31/

**Jenkins build:**
https://ddt-jenkins.prod.flightstats.io/job/run-hub-development-integration/177/

**Explanation for changes:**
this is a test file which is writtten in JS.

1.**Sanitize location:**:
Before using or logging the location header, we sanitize it using encodeURIComponent to ensure that no sensitive information is exposed.

2.**Log Sanitized Data Only:**
The latestKey is derived from the sanitized location, and only this sanitized version is logged.

3.**Additional Safety Check:**
Added a check to ensure that location is defined before proceeding with encoding and substring operations.
By implementing these changes, we ensure that the handling of the location header and the extraction of the latestKey are performed securely, mitigating the risk of privacy violations and inadvertent exposure of sensitive data.

### CODE REVIEW

**Review by harinadh-dasari — APPROVED** [review]
LGTM

File: src/test/javascript/continuous/verify_latest_spec.js [hunk 1]
```diff
@@ -80,9 +80,11 @@ describe(__filename, function () {
                 item.empty = true;
             } else if (statusCode === 303) {
                 const location = fromObjectPath(['headers', 'location'], res);
+                // Ensure location does not contain sensitive information before logging or processing
+                const sanitizedLocation = location ? encodeURIComponent(location) : 'unknown';
                 const strLength = channelUrl.length + name.length + 2;
-                const latestKey = location.substring(strLength);
-                console.log('latestKey ', latestKey, location);
+                const latestKey = sanitizedLocation.substring(strLength);
+                console.log('latestKey (sanitized) ', latestKey);
```
Comment (allustag) [inline src/test/javascript/continuous/verify_latest_spec.js:87]: Have to check what will be value of `latestKey`, if `location` is set to '_unknown_'

Comment (Haripriya99j) [reply src/test/javascript/continuous/verify_latest_spec.js:87]: When location is missing or unknown, we handle it by assigning "null" to that latest key.
unknown log message will assign this value as null only, lastly the item is resulting as "null"

Comment (allustag) [reply src/test/javascript/continuous/verify_latest_spec.js:87]: `fromObjectPath(['headers', 'location'], res);` if this code at line#82 returns null, then `sanitizedLocation` will be set to '_unknown_'

**Review by allustag — APPROVED** [review]

**Pull request merged**